### PR TITLE
Fixes #352 Validate yaml file and ignore invalid input instead of crashing

### LINKED ISSUE #352: Handle error in yaml file with a error message instead of crashing.
### This is a
- [ ] Bug report
- [ ] Feature request

### Current behaviour

Currently the script crashes with a traceback.

### Expected behaviour

The script should print a message to the console and exit.

### PR DESCRIPTION
**Pull request by pri22296**
## Description

- Handle various errors in the format of yaml file and instead of failing build, provide a message to stderr and continue processing the file.
- Converted the module to a package. This was done as the code was getting too big. It was getting tough to maintain.
- Added validation system for various attributes(if a value is not valid, it would be removed from the configuration instead of passing the value forward to the script which may ultimately cause build failure or other side effects.

## Related Issue

Fixes #352

## How Has This Been Tested?

Deployment: https://yaydocx.herokuapp.com

## Screenshots (if appropriate):
screenshot 212
screenshot 213

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
@pri22296 once I got an error on empty yaml file. if would be great if you handle that also

**Comment by pri22296** [comment]
@sch00lb0y Good point. Thanks for pointing out.

**Comment by pri22296** [comment]
@imujjwal96 @sch00lb0y Please review

**Comment by sch00lb0y** [comment]
@pri22296 suggestion: is it possible to validate giturl, twitter URL because we not validating in the script it will be good if we validate here. the code looks good to me.

**Comment by imujjwal96** [comment]
Still a `WIP`, please add the tag

**Comment by pri22296** [comment]
@sch00lb0y @imujjwal96 Updated

File: modules/scripts/config/validation.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    mimetype = mimetypes.guess_type(value)[0]
+    if mimetype is None:
+        return False
+    else:
+        return mimetype.startswith('image')
+
+
+def validate_subproject(value):
+    regex = '(http|https)://(www.|)github.com/([\w\d\.]+)/([\w\d\.]+)(.git|)'
+    return re.match(regex, value['url']) != None
```
Comment (sch00lb0y) [inline modules/scripts/config/validation.py]: validate twitter url as well :smile:

Comment (pri22296) [inline modules/scripts/config/validation.py]: As far as i remember, there's no twitter url option in the yaml file.

Comment (sch00lb0y) [inline modules/scripts/config/validation.py]: I totally forgot. :sweat:

**Pull request merged**